### ISSUE TITLE
Improving experiment configuration

### ISSUE THREAD

**Issue by demmerichs** [open-mmlab/OpenPCDet#1278]
Hi,

I am working with this toolbox now for quite some time and want to thank everyone involved for this great resource. I would like to contribute one feature as its missing annoys me quite a bit, but before starting to work on it, I would like to get some feedback on the general idea and if such a PR would be accepted:

My "problem":
I am a researcher and I guess this toolbox is mostly used by DL researchers. Trying out different experiment setups in an organized manner is important. The way, the experiment configuration is currently set up, suggests generating a new configuration file for each change, even if it is just swapping the loss function or changing a loss weight or layer size. When using the `train.py` script, you can use `--set MODEL.MLP [32,64]` as an optional parameter that changes configuration values on the fly, however, the output directory that is created simply copies the original configuration file over, without the changes made through the optional parameter settings during the call of the train script. This makes post-processing, e.g. inference on a test set, a bit tricky, as you need to manually provide the set configuration changes again to the `test.py` script or change the copied base configuration by hand.

My solution:
I am personally using my PyPi package [`runcon`](https://pypi.org/project/runcon/) now every time when working with OpenPCDet. It allows me to easily do smaller and also grouped config changes while autogenerating descriptive labels for my output directories. In addition, I not only copy the base configuration file but also save the change configuration file used during training into the output directory, which then can be directly used for post-processing scripts.

I would be thankful if anyone could chime in and give feedback on my idea, and if acceptable, would guide me through getting a PR of this kind into this repository.
